airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretPersistence.java

```diff
@@ -68,4 +68,12 @@ static Optional<SecretPersistence> getEphemeral(final DSLContext dslContext, fin
     }
   }
 
+  static SecretsHydrator getDataPlaneSecretsHydrator(final Configs configs) {
```

Comment:
why is this a separate method? would it make sense to just do the appropriate validations in `getSecretsHydrator`? If there is secret-critical things that depend on the developer knowing that these 2 methods exist and choosing the right one, that seems a bit risky. Can we remove the choice and just guarantee that the one method always does the right thing?

Reply:
Discussion continued in child PR here: https://github.com/airbytehq/airbyte/pull/15904/files#r953196612